app/controllers/api/service_templates_controller.rb

```diff
@@ -4,5 +4,13 @@ class ServiceTemplatesController < BaseController
     include Subcollections::Tags
     include Subcollections::ResourceActions
     include Subcollections::ServiceRequests
+
+    def create_resource(_type, _id, data)
+      raise 'Must provide a type' unless data['type']
+      type = data['type'].constantize
```

Comment:
Here we have both `_type` and `type`. Consider to rename either one to something more meaningful.